### DISCUSSION TITLE
Models API endpoint returns 0 models for normal user

### DISCUSSION THREAD

**Discussion by NiLon** [open-webui/open-webui#7107] [General]
# Bug Report

## Environment

- **Open WebUI Version:** v0.4.1

## Expected Behavior:

To get list of models that user has access for.

## Actual Behavior:

0 models in web page and api endpoint

## Description

**Bug Summary:**
Whenever user is "user" they get zero models no matter what is allowed. Swapping them to admin allows them to see models. 
User can however see list of models in the new chat dropdown, but not in models page.
HTTP url is being used, seeing something being worked on HTTPS side might have caused a side effect.
Note sure if this worked on 0.4.0 but before that it did for sure.

**Comment by tjbck** [comment]
<img width="890" alt="image" src="https://github.com/user-attachments/assets/5a220250-2285-4bff-b5ef-445b4b5728cb">

https://github.com/open-webui/open-webui/releases/tag/v0.4.0

**Reply by NiLon** [reply]
That's great. There are two places to modify, in the model page it has to be enabled and then edited so that it is public or allowed for group that the user must be in. Then second is in admin model listing with similar option. These do nothing at all. It doesn't matter if they are all public or private with user in group. The only setting that does anything is the admin/user flag.

**Reply by fawwazanvilen** [reply]
yeah, the previous behavior was perfect for my use case. i share my open-webui instance with my friends through openrouter.ai api and i encourage them to explore the models that are available there.

now the only way to conveniently do this is by assigning them as admin - which is undesirable. the private by default behavior is nice, but there should be a menu/checklist ui where you can select models and assign them to be public/private.

**Reply by chan-a** [reply]
I also prefer the previous behavior. I have no issue with allowing "users" to access all models (which are loaded from OpenRouter). To do so at the moment, I either have to make them "admins" (not preferred), or go through the 200+ model to set them as public (not realistic).

**Reply by sczhengyabin** [reply]
Please, at least, add a batch process approach to those models' permissions.

**Reply by simonfrey** [reply]
Yeah, this behavior broke the ui. Does not make sense for all usecases, would be good if we can configure it via env var

**Comment by simonfrey** [comment]
Added a PR to allow for global overwrite to allow all models for all users again: https://github.com/open-webui/open-webui/pull/7459

**Reply by tjbck** [reply]
`BYPASS_MODEL_ACCESS_CONTROL` has been added to dev!
